[fix] #7 장소 세부정보 반환 오류 패치

dong-yxxn opened the pull request:
### ✨ 작업 내용
- 기존 PlaceService에서 video가 null일때 예외처리를 적용했습니다.
- video가 Null이면, PlaceDetailInfo의 videoUrl과 influencerName을 빈 문자열로 처리했습니다.
- place-video간 일대다 관계를 placeService 로직에 반영했습니다.
- 관련된 테스트 코드 수정했습니다.
---

### ✨ 참고 사항

---

### ⏰ 현재 버그

---

### ✏ Git Close

dong-yxxn committed:
[refactor] QueryDSL 적용

- QueryDsl을 적용하기위해 customRepo를 작성하였습니다.

관련 이슈: #7

dong-yxxn committed:
Merge remote-tracking branch 'origin/weekly/6' into feat/#7-place

dong-yxxn committed:
[fix] PlaceService에서 video null 처리 구현

- placeService에서 video가 조회되지 않는 오류를 예외처리하였습니다.

관련 이슈: #7

BaeJunH0 commented on src/main/java/team7/inplace/place/application/PlaceService.java:
```diff
@@ -38,95 +39,103 @@ public Page<PlaceInfo> getPlacesWithinRadius(
-                        + "Place Id가 존재하지 않습니다."));
-        Video video = videoRepository.findByPlaceId(placeId)
-                .orElseThrow(() -> new IllegalArgumentException("PlaceService.getPlaceDetailInfo(): "
+            .orElseThrow(() -> new IllegalArgumentException("PlaceService.getPlaceDetailInfo(): "
```
Error 코드 작성해서 해봅시다용

BaeJunH0 commented on src/main/java/team7/inplace/place/application/PlaceService.java:
```diff
@@ -38,95 +39,103 @@ public Page<PlaceInfo> getPlacesWithinRadius(
+        Video video = null;
+        Influencer influencer = null;
+        try {
+            video = videoRepository.findByPlaceId(placeId)
+                .orElseThrow(
+                    () -> new IllegalArgumentException("PlaceService.getPlaceDetailInfo(): "
                         + "Place Id가 존재하지 않습니다."));
-        return PlaceDetailInfo.from(place, video.getInfluencer(), video);
+            influencer = video.getInfluencer();
+        } catch (IllegalArgumentException e) {
+
+        }
```
try catch가 필요할까요? Exception이 아니라 Error로 처리하면 사용하지 않아도 되지 않을까요?

sanghee0820 replied:
Exception Handler에서 값을 받도록 해야할 것 같습니다! 다른분들이 작성하신 코드 확인해보세요!

BaeJunH0 commented on src/main/java/team7/inplace/place/persistence/PlaceCustomRepository.java:
```diff
@@ -1,2 +1,21 @@
-package team7.inplace.place.persistence;public interface PlaceCustomRepository {
+package team7.inplace.place.persistence;
+
+public interface PlaceCustomRepository {
+
+//    Page<Place> findPlacesByDistance(
+//        @Param("longitude") String longitude,
+//        @Param("latitude") String latitude,
+//        Pageable pageable);
+//
+//    Page<Place> findPlacesByDistanceAndFilters(
+//        @Param("topLeftLongitude") String topLeftLongitude,
+//        @Param("topLeftLatitude") String topLeftLatitude,
+//        @Param("bottomRightLongitude") String bottomRightLongitude,
+//        @Param("bottomRightLatitude") String bottomRightLatitude,
+//        @Param("longitude") String longitude,
+//        @Param("latitude") String latitude,
+//        @Param("categories") List<String> categories,
+//        @Param("influencers") List<String> influencers,
+//        Pageable pageable);
+
 }
```
?

BaeJunH0 commented on src/main/java/team7/inplace/place/persistence/PlaceCustomRepositoryImpl.java:
```diff
@@ -1,2 +1,94 @@
+//            ).orderBy(distanceExpression.asc())
+//            .offset(pageable.getOffset())
+//            .limit(pageable.getPageSize())
+//            .fetch();
+//
+//        return new PageImpl<>(places, pageable, places.size());
+//    }
+//
+//    private BooleanExpression withinBoundary(QPlace place, double topLeftLongitude,
+//        double topLeftLatitude,
+//        double bottomRightLongitude, double bottomRightLatitude) {
+//        NumberTemplate<Double> longitude = Expressions.numberTemplate(Double.class,
+//            "CAST({0} AS DOUBLE)", place.coordinate.longitude);
+//        NumberTemplate<Double> latitude = Expressions.numberTemplate(Double.class,
+//            "CAST({0} AS DOUBLE)", place.coordinate.latitude);
+//
+//        return longitude.between(topLeftLongitude, bottomRightLongitude)
+//            .and(latitude.between(bottomRightLatitude, topLeftLatitude));
+//    }
+//
+//    private BooleanExpression placeCategoryIn(List<String> categories) {
+//        if (categories == null) {
+//            return null;
+//        }
+//
+//        List<Category> enumCategories = categories.stream()
+//            .map(Category::valueOf)
+//            .collect(Collectors.toList());
+//
+//        return QPlace.place.category.in(enumCategories);
+//    }
+//
+//    private BooleanExpression placeInfluencerIn(List<String> influencers) {
+//        if (influencers == null) {
+//            return null;
+//        }
+//
+//        return QInfluencer.influencer.name.in(influencers);
+//    }
 }
```
나중에 쓰려고 올린건가요? 다음부터는 커밋하지 말아주세용

BaeJunH0 reviewed (approved):
한 번 읽어봐주세요~

sanghee0820 reviewed (changes requested):
일단은 Exception 부분만 수정해주시면 될 것 같습니다.

dong-yxxn committed:
[refactor]: 장소 세부정보 조회에서 로직 수정

- 장소세부정보 조회에서 video 정보가 있으면 가져오고 없으면 빈문자열()만 들어가도록 수정하였습니다.

관련 이슈: #7

dong-yxxn committed:
[refactor]: place-video간 일대다 관계 반영

- VideoRepository: findByPlaceId시, Video 여러개 조회되는 문제로 List로 return하도록함
- PlaceService: video 여러개 조회될시에 제일 첫번째 video만 반영

- 관련하여 테스트 코드 추가함

관련 이슈: #7

sanghee0820 reviewed (approved):
임시로 approve하겠습니다.
